book/src/developers/designs/profile_display.md

```diff
@@ -0,0 +1,136 @@
+## User Settings Display Web UI
+
+We need a usable centralized location for users to view and manage their own user settings.
+User settings do not envelop credentials, these have their own flow as they are also used in user setup.
+
+ - Writing new values requires a writable session -> we will make the user reauthenticate to obtain a temporary profile-update-token when they want to update their user settings.
+ - The UI must display and make editable the following categories:
+   - user attributes
+   - user ssh public-keys
+ - The UI must display:
+   - user credential status as per [credential-display.rst](credential-display.rst)
+   - user groups
+
+### User attributes
+These consist of: 
+ - username
+ - displayname
+ - legal name
+ - email address
+In the future:
+ - picture
+ - zoneinfo/timezone
+ - locale/preferred language
+ - other business related attributes: address, phone number, ...
+
+#### Displaying attributes
+Attributes should be displayed with
+ - their descriptive name with tooltips or links if the name may be confusing to non IT people.
+ - their current value OR if not set: some clear indication that the attribute is not set.
+ - A method to edit the attribute if it is editable
+
+#### Editing attributes
+Users must be able to edit attributes individually.
+Users should be able to see their changes before saving them. 
+  E.g. via a popup that shows the old vs new value asking to confirm.
+
+#### TODO: Personal Identifiable Information attributes (currently we don't have these attributes)
+Certain information should not be displayed in the UI without reauthentication:
+ - addresses
+ - phone numbers
+ - personal emails
+ - birthdate
+
+### SSH public keys
```

Comment:
I think ssh keys and unix password will move to the cred update flow.

Reply:
Oh okay, I think that makes sense

Reply:
Should we reuse the cred update page content for credential status section? They seem mostly similar other than the ability to update them. Then again, I'm also not sure if there's a need to hide the navbar for credentials reset page.

Reply:
The credential reset page needs to be outside of the navbar because it happens in a credential update transaction, which also requires a re-auth. That's why we swap to a separate page. I think if we kept the navbar, we'd need a way to implicitly cancel the cred update on navigate away.

Reply:
It makes sense that reauth doesn't have navbar because you go to the auth screen, but after that on the cred update page it shouldn't be necessary other than for artificially blocking users from navigating to other pages. Do we handle when the user doesn't complete the credential update by canceling implicitly now? My impression is that we don't (navigating away and going back to the update flow and the flow is still alive) and I know from my experience of poking around that it does expire eventually.

The other part of this is me being a UI guy... we should be able to update creds in the app itself without being in a modal screen. We don't need to couple UI and logic. If you agree I will be happy to look into this direction :)

Reply:
We don't actually, you're right. We just time-out the credential update session in the background if they navigate away. So we could probably change the UI and just rely on that background garbage collect instead if you think the UI can be improved greatly :)

Reply:
I will also see if I can cancel when the user navigates away... for science